ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -194,6 +197,53 @@ def _llmobs_set_meta_tags_from_chain(
             except TypeError:
                 log.warning("Failed to serialize chain output data to JSON")
 
+    def _llmobs_set_meta_tags_from_similarity_search(
+        self,
+        span: Span,
+        input_query: str,
+        output_documents: Union[List[Any], None],
+        error: bool = False,
+        is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "similarity_search")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        input_tag_key = INPUT_VALUE if is_workflow else INPUT_MESSAGES
+        output_tag_key = OUTPUT_VALUE if is_workflow else OUTPUT_DOCUMENTS
+
+        if input_query is not None:
+            try:
+                formatted_inputs = self.format_io(input_query)
+                if isinstance(formatted_inputs, str):
+                    span.set_tag_str(input_tag_key, formatted_inputs)
+                else:
+                    span.set_tag_str(input_tag_key, json.dumps(self.format_io(input_query)))
+            except TypeError:
+                log.warning("Failed to serialize similarity input query to JSON")
+        if error:
+            span.set_tag_str(output_tag_key, "")
+        elif isinstance(output_documents, list):
+            if is_workflow:
+                span.set_tag_str(output_tag_key, json.dumps(output_documents))
+            else:
+                try:
+                    documents = [
+                        Document(
+                            id=d.id,
+                            text=d.page_content,
```

Comment:
@Yun-Kim Is it okay that page content isn't truncated ?

Reply:
Currently yes it's fine, we don't include any truncation in our document support. We will perform truncation/obfuscation at the trace processing level, no need for this at the integration level.